models/classes/eventHandler/ResultTransmissionEventHandler/AsynchronousResultTransmissionEventHandler.php

```diff
@@ -66,6 +68,13 @@ private function packVariables(array $variables): array
     {
         return array_map(static function ($record) {
             if (is_object($record)) {
+                if ($record instanceof ResponseVariable) {
+                    $value = $record->getValue();
+                    if ($value instanceof QtiFlysystemFile) {
+                        //clear cached dependency injections
+                        $record->setValue(new QtiFlysystemFile($value->getFileSystemId(), $value->getPrefix()));
```

Comment:
And then here instead of re-creating an instance, you may call `$record->setServiceLocator(new \oat\oatbox\service\ServiceManager(\common_ext_ConfigDriver::singleton()))`

Reply:
all this recreation just for clearing filesytem client which cached in `QtiFlysystemFile`